src/str_extract.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import os, sys, re
+import argparse
+
+def strextract(dir, suffix=None, path=False, all=False):  
+    # Expression régulière pour trouver des chaînes littérales entre guillemets simples ou doubles
+    # compiled_re = re.compile(r"^.*'.*'.*"'"'r".*"'"'r"$", re.IGNORECASE)
+    regex_chain = re.compile(r'(["\'])(.*?)\1', re.IGNORECASE)
+    # Parcourir récursivement le répertoire
+    for root, directorys, filename in os.walk(dir):
```

Comment:
Rename `directorys` into `_` to show that it does not interest you
Rename `filename` into `filenames` ot `filename_list` to express that it's a list and that it contains file names as strings

Reply:
I actually need `directorys`, it's the `root` bit that i don't need. 
And there is a mistake in the name, I'll rename it as `directories`

Reply:
Then I missed an error: line 14, it should be `file_path = os.path.join(root, name_file)`
You don't need `directorys` but you do need `root`,